src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -158,6 +158,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
      */
     public function buildView(FormView $view, FormInterface $form, array $options)
     {
+        $choiceTranslationDomain = $options['choice_translation_domain'];
+        $translationDomain = $options['translation_domain'];
+
+        if ($view->parent) {
+            if (null === $choiceTranslationDomain) {
+                $choiceTranslationDomain = $view->parent->vars['translation_domain'];
```

Comment:
it should use the translation domain of the *current* form when set to null, not of the parent one (the parent value is read for the ``translation_domain`` inheritance